Source/Shared/BPUtils.h

```diff
@@ -69,11 +69,10 @@ typedef NS_ENUM(int, BPKind) {
 + (void)printInfo:(BPKind)kind withString:(NSString *)fmt, ...;
 
 /*!
- @discussion print a message to stderr.
- @param kind one of the levels in BPKind
- @param fmt a format string (a la printf), followed by the var args.
+ @discussion get an NSError *
+ @param fmt a format string (a la printf), followed by var args.
  */
-+ (void)printError:(BPKind)kind withString:(NSString *)fmt, ...;
++ (NSError *)BPError:(NSString *)fmt, ... ;
```

Comment:
The main reason I've seen for using a macro for errors is when you use the compiler internals like `__function__` or `__line__` to get information for the error. We could add that and then it'd make sense to have it be a macro. 

Speed on errors is rarely a concern, you're failing anyway... why be fast?  😄 

Let me think about how to preserve the variadic arguments (which I like) in a macro. I've done it before, but not in Objective-C.